util/migration-rocksdb/src/lib.rs

```diff
@@ -98,39 +98,39 @@ impl Batch {
 /// A generalized migration from the given db to a destination db.
 pub trait Migration: 'static {
```

Comment:
Out of scope:

the entire trait doesn't have to be 'static:

```rustc
diff --git a/util/migration-rocksdb/src/lib.rs b/util/migration-rocksdb/src/lib.rs
index 8a9c23f88..bef4b885e 100644
--- a/util/migration-rocksdb/src/lib.rs
+++ b/util/migration-rocksdb/src/lib.rs
@@ -96,7 +96,7 @@ impl Batch {
 }
 
 /// A generalized migration from the given db to a destination db.
-pub trait Migration: 'static {
+pub trait Migration {
 	/// Number of columns in the database before the migration.
 	fn pre_columns(&self) -> u32 { self.columns() }
 	/// Number of columns in database after the migration.
@@ -111,7 +111,7 @@ pub trait Migration: 'static {
 }
 
 /// A simple migration over key-value pairs of a single column.
-pub trait SimpleMigration: 'static {
+pub trait SimpleMigration {
 	/// Number of columns in database after the migration.
 	fn columns(&self) -> u32;
 	/// Version of database after the migration.
@@ -217,7 +217,7 @@ impl Manager {
 	}
 
 	/// Adds new migration rules.
-	pub fn add_migration<T>(&mut self, migration: T) -> io::Result<()> where T: Migration {
+	pub fn add_migration<T: 'static>(&mut self, migration: T) -> io::Result<()> where T: Migration {
 		let is_new = match self.migrations.last() {
 			Some(last) => migration.version() > last.version(),
 			None => true,

```
I can fix after this PR if you don't want to do it but I don't want introduce merge conflicts :P

Reply:
Sounds like a plan! Do you mind adding an issue and whoever gets there first etc? :)

Reply:
Sure, #11342